src/client/src/database.rs

```diff
@@ -61,46 +66,61 @@ impl Database {
         self.schema = schema.into();
     }
 
-    pub async fn insert(&self, request: InsertRequest) -> Result<Output> {
-        self.do_get(Request::Insert(request)).await
+    pub async fn insert(&self, request: InsertRequest, ctx: FlightContext) -> Result<Output> {
```

Comment:
Instead of giving each method a context argument, can be bind it to `Database`?

Reply:
IMHO, there are 3 level(actually 4 including grpc client level) of abstraction here
1. database level, bound to a certain database with fixed `catalog` and `schema`. Theoretically it can be reused by multiple connections from different user
2. connection level, bound to a certain connection create by one user, much like a mysql's session
3. request level, carrying per request parameter 

Currently `Database` is created with `catalog` and `schema`, as its name suggests, playing the database level role. If we bind it with auth info(username and password, etc.), it would be actually a connection level struct

But it can make some sense in  'connecting to a database, authenticate, and return a database handle` way, and since it's an public api, it possibly won't be wrapped to create connection level abstraction again. So I'm good with it. It's also easy to modify to add auth scheme within `Database` if we can all agree on this

please take a look too @MichaelScofield

Reply:
IIRC, grpc does not expose connection state to upper layer and it tries to be a stateless transport. So we might not be able to track if a connection has been authenticated like mysql or postgres. So I agree we need request-level authentication.

However, when creating a `Database` instance, `catalog` and `schema` are provided, which means the instance is bound to a particular database. I think authentication information should always go with database. There is little chance that different users shares same connection, but connected to same database.

If the connection can be shared by different users, we better design it to be shared among databases as well.

Reply:
I think it's better to store `FlightContext` in `Database`. It's easier to use. If you are worry about the context should be consistent with `catalog` and `schema`, you can create a struct to hold the 3 variants, change the visibility or add some helper methods to make them changed together.

Reply:
@sunng87 @MichaelScofield @Fengys123  PTAL